extra_geom/detectors.py

```diff
@@ -854,6 +854,170 @@ def to_distortion_array(self, allow_negative_xy=False):
         return super().to_distortion_array(allow_negative_xy)
 
 
+class AGIPD_500KGeometry(DetectorGeometryBase):
+    """Detector layout for AGIPD-500k
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+
+    You won't normally instantiate this class directly:
+    use one of the constructor class methods to create or load a geometry.
+    """
+    detector_type_name = 'AGIPD-500K2G'
+    pixel_size = 2e-4  # 2e-4 metres == 0.2 mm
+    frag_ss_pixels = 64
+    frag_fs_pixels = 128
+    expected_data_shape = (8, 512, 128)
+    n_modules = 8
+    n_tiles_per_module = 8
+
+    @classmethod
+    def from_center(cls, center, asic_gap=2, panel_gap=(16, 30), unit=pixel_size):
+        """Generate an AGIPD-500K2G geometry from center position.
+
+        This produces an idealised geometry, assuming all modules are perfectly
+        flat, aligned and equally spaced within their quadrant.
+
+        The center position is given in pixel units, referring to the first
+        pixel of the first module, corresponding to data channels 0.
+
+        The origin of the coordinates is the bottom-right corner of the
+        detector. Coordinates increase upwards and to the left (looking along
+        the beam).
```

Comment:
This is a bit confusing. It appears to be saying that you give the location of M0 with respect to the bottom beam-right corner (by M6), but that's not the case.

We must be either giving the location of M0 with respect to the centre (the point where a ray from the sample would hit perpendicular to the detector plane) or vice versa - the position of the centre with respect to some fixed point on the detector hardware. I think the implementation is the former, but calling this `from_center` suggests the latter.

Reply:
Thanks, I updated the text, hope it's clearer.